### ISSUE TITLE
Screen readers are announcing an incorrect menu item count on dropdown

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3552]
## Describe the bug

Screen readers are announcing an incorrect menu item count on dropdown. This may be caused by some items missing role="menuitem".

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using Mac OS X 10_14_4/VoiceOver/Safari Version 12.1.1 (14607.2.6.1.1)
- Activate the Dropdown Demos link under the Dropdown heading
- Activate the Dropdown 1 link under the 'Positioning Demo' heading
- Observe that VoiceOver announces that the menu has 6 items, but there are only 4 visible.

## Expected behavior

There should be proper semantics.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

Follow the example at
http://whatsock.com/tsg/Coding%20Arena/ARIA%20Menus/Vertical%20(Internal%20Content)/demo.htm
